dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/DatadogCore.kt

```diff
@@ -371,7 +358,9 @@ internal class DatadogCore(
     }
 
     override fun getDatadogContext(): DatadogContext? {
-        return contextProvider?.context
+        return coreFeature.contextExecutorService
+            .submitSafe("getDatadogContext", internalLogger, Callable { contextProvider?.context })
```

Comment:
is this going to block if the queue is not ready ?

Reply:
Yes, it will. This call is blocking on purpose.